### PR TITLE
Scripting: Remove file scripts

### LINKED ISSUE #21798: Should we remove file-based scripts and templates?
We have largely removed support for file-based configuration (eg mappings, settings, index templates) in favour of storing this info in the cluster.

For scripts and mustache templates, we allow them to be passed inline or to be stored in the cluster state.  We used to be concerned about locking down scripts in Groovy etc, but with Painless we have safe scripting. Are file based scripts now redundant?

The one upside of files is that scripts can be written with new lines and formatting, while with inline/stored scripts they have to be passed in in a single line of JSON with newlines escaped.  However this could be solved by improving Console to do the multi->single line conversion automatically.

Any other reasons for keeping file scripts?

### PR DESCRIPTION
**Pull request by rjernst**
This commit removes file scripts, which were deprecated in 5.5.

closes #21798

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java [hunk 1]
```diff
@@ -225,19 +176,11 @@ public void testDefaultBehaviourFineGrainedSettings() throws IOException {
             deprecate = true;
         }
         buildScriptService(builder.build());
-        createFileScripts("dtest");
 
         for (ScriptContext scriptContext : scriptContexts) {
             // only file scripts are accepted by default
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java]: This comment can go.

File: core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java [hunk 2]
```diff
@@ -306,12 +249,11 @@ public void testFineGrainedSettings() throws IOException {
         }
 
         buildScriptService(builder.build());
-        createFileScripts("expression", "mustache", "dtest");
 
         for (ScriptType scriptType : ScriptType.values()) {
             //make sure file scripts have a different name than inline ones.
             //Otherwise they are always considered file ones as they can be found in the static cache.
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java]: These comments about file scripts can go.

File: core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java [hunk 3]
```diff
@@ -306,12 +249,11 @@ public void testFineGrainedSettings() throws IOException {
...
-            String script = scriptType == ScriptType.FILE ? "file_script" : "script";
+            String script = "script";
             for (ScriptContext scriptContext : this.scriptContexts) {
                 //fallback mechanism: 1) engine specific settings 2) op based settings 3) source based settings
                 Boolean scriptEnabled = engineSettings.get(dangerousScriptEngine.getType() + "." + scriptType + "." + scriptContext.getKey());
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java]: It won't let me review line 278 where there is still an assertion about file scripts.

File: core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java [hunk 4]
```diff
@@ -71,7 +70,6 @@
     private static final Map<ScriptType, Boolean> DEFAULT_SCRIPT_ENABLED = new HashMap<>();
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java]: There's a field up there `scriptsFilePath` that I think can go?

File: core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java [hunk 5]
```diff
@@ -56,7 +56,6 @@
 //TODO: this needs to be a base test class, and all scripting engines extend it
 public class ScriptServiceTests extends ESTestCase {
 
-    private ResourceWatcherService resourceWatcherService;
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java]: There are imports up above that can be removed.

File: core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java [hunk 6]
```diff
@@ -225,19 +176,11 @@ public void testDefaultBehaviourFineGrainedSettings() throws IOException {
             deprecate = true;
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java]: I think this can go?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left some comments, nothing major.

**Comment by jasontedor** [comment]
Also, I *think* there are some packaging tests that set up a file script?

**Comment by rjernst** [comment]
@elasticmachine retest this please

**Comment by rjernst** [comment]
@jasontedor Packaging tests are updated.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**